catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePollerRunner.java

```diff
@@ -159,7 +159,14 @@ public void unbind(Source source) {
         LOGGER.debug("Unbinding source [{}]", source);
         if (source != null) {
             sources.remove(source);
-            sourceStatusThreadLocks.remove(cachedSources.remove(getSourceKey(source)));
+            SourceKey key = getSourceKey(source);
+            CachedSource cachedSource = null;
```

Comment:
I don't know that there is much value in logging here. If there is a problem here then it has already been removed.